src/main/kotlin/com/pubnub/api/v2/subscriptions/SubscriptionOptions.kt

```diff
@@ -0,0 +1,80 @@
+package com.pubnub.api.v2.subscriptions
+
+import com.pubnub.api.models.consumer.pubsub.PNEvent
+import com.pubnub.internal.v2.subscription.Filter
+import com.pubnub.internal.v2.subscription.ReceivePresenceEvents
+
+/**
+ * SubscriptionOptions is a mechanism used for supplying optional modifiers for subscriptions.
+ *
+ * The options currently available in the PubNub client are:
+ * * [SubscriptionOptions.filter]
+ * * [ChannelOptions.receivePresenceEvents]
```

Comment:
My first impression is that [ChannelOptions.receivePresenceEvents] looks strange here.  "subscription" function takes SubscriptionOptions ( override fun subscription(options: SubscriptionOptions): Subscription ) and it is possible to pass ChannelOptions to it.

Wouldn't it be more clear to have  [SubscriptionOptions.receivePresenceEvents]?

For me it is ok to treat "receivePresenceEvents" as SubscriptionOptions instead of ChannelOptions.

Reply:
I simplified all of this to just two possible function calls